core/lib/zksync_core/src/native_token_fetcher/mod.rs

```diff
@@ -1,19 +1,18 @@
-use std::{cmp::min, sync::Arc, time::Duration};
+use std::{cmp::min, sync::Arc};
 
 use anyhow::Context;
 use async_trait::async_trait;
 use hex::ToHex;
 use metrics::atomics::AtomicU64;
-use tokio::{
-    sync::{watch, OnceCell},
-    task::JoinHandle,
-};
+use tokio::sync::Mutex;
```

Comment:
Side comment: mutexes from std are perfectly fine to use in an async context, and generally they are more performant than `tokio` mutex. Usually, you only need `tokio` mutex if you need to hold a guard across an `await` point (and even in such situations, it's often a code smell). See the [tokio docs for context](https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html#which-kind-of-mutex-should-you-use).

This is a minor thing though and can be kept as is.